graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,303 @@
+            msg = "R must be an integer > 0"
+            raise ValueError(msg)
+
+    def fit(self, voi, seeds=[], A=None, B=None):
+        """
+        Fits the model to two graphs.
+
+        Parameters
+        ----------
+        voi: int
+            Vertex of interest (voi)
+
+        seeds: list
+            List of length two `[seedsA, seedsB]` where first element is
+            the seeds associated with adjacency matrix A
+            and the second element the adjacency matrix associated with B, note
+            `len(seedsA)==len(seedsB)`. The elements of `seedsA` and `seedsB` are
+            vertices which are known to be matched, that is, `seedsA[i]` is matched
+            to vertex `seedsB[i]`.
+
+        A: 2d-array, square
+            Adjacency matrix of `A`, the graph where voi is known
+
+        B: 2d-array, square
+            Adjacency matrix of `B`, the graph where voi is not known
+
+        Returns
+        -------
+        self: A reference to self
+        """
+        if A is None or B is None:
+            msg = "Adjacency matricies must be passed"
+            raise ValueError(msg)
+        elif len(A.shape) != 2 or len(B.shape) != 2:
+            msg = "Adjacency matrix entries must be square"
+            raise ValueError(msg)
+        elif A.shape[0] != A.shape[1] or B.shape[0] != B.shape[1]:
+            msg = "Adjacency matrix entries must be square"
+            raise ValueError(msg)
+
+        if len(seeds) == 0:
+            print("Must include at least one seed to produce nomination list")
+            return None
+        if len(seeds) != 2:
+            msg = "List must be length two, with first element containing seeds \
+                  of A and the second containing seeds of B"
+            raise ValueError(msg)
+
+        seedsA = seeds[0]
+        seedsB = seeds[1]
+
+        if len(seedsA) != len(seedsB):
+            msg = "Must have the same number of seeds for each adjacency matrix"
+            raise ValueError(msg)
+        if len(seedsA) == 0:
+            print("Must include at least one seed to produce nomination list")
+            return None
+
+        voi = np.reshape(np.array(voi), (1,))
+
+        # get reordering for Ax
+        nsx1 = np.setdiff1d(np.arange(A.shape[0]), np.concatenate((seedsA, voi)))
+        a_reord = np.concatenate((seedsA, voi, nsx1))
+
+        # get reordering for B
+        nsx2 = np.setdiff1d(np.arange(B.shape[0]), seedsB)
+        b_reord = np.concatenate((seedsB, nsx2))
+
+        AA = A[np.ix_(a_reord, a_reord)]
+        BB = B[np.ix_(b_reord, b_reord)]
+
+        seeds_reord = np.arange(len(seedsA))
+        voi_reord = len(seedsA)
+
+        subgraph_AA = np.array(
+            _get_induced_subgraph_list(
+                AA, self.order_voi_subgraph, voi_reord, mindist=1
+            )
+        )
+
+        voi_reord = np.reshape(np.array(voi_reord), (1,))
+
+        Sx1 = np.intersect1d(subgraph_AA, seeds_reord)
+
+        if len(Sx1) <= 0:
```

Comment:
this should raise a valueerror. in general you should never really print anything. if something happens that shouldn't  happen, you should raise an error.

Reply:
<img width="1194" alt="Screen Shot 2020-12-09 at 1 06 59 PM" src="https://user-images.githubusercontent.com/32661399/101669076-0e175100-3a17-11eb-92d2-3883d0ce7d78.png">

In this algorithm in step 1 there is a stopping condition. This is what this if statement is checking. It would be hard for users to know if this stopping condition will be hit before the algorithm runs especially if the graphs are big and there aren't very many seeds. I think the only sensible way to figure this out would to be run 

`
subgraph_AA = np.array(
    _get_induced_subgraph_list(
        AA, self.order_voi_subgraph, voi_reord, mindist=1
    )
)
`

However, this is exactly what is in the code, so if we required the user to run this before they ran VNviaSGM, then there would be a redundancy. 

I guess my concern is that reaching this condition should happen sometimes, and if it does happen the user shouldn't think it is a problem with his inputs. In this condition the algorithm just stops because there are no seeds close enough to the vertex of interest.

Reply:
Fair, but we still don't want fit to return `None`. I'm guessing this happens sometimes and you don't this occurrence to stop a bunch of code from running. In which case, I say we raise a warning, set `nomination_list_` equal to something like `None`, and return self there so that we don't go through the rest of the algorithm. Sound good?

Reply:
Sounds good, Ill do that now.